src/ui/public/index_patterns/_get_template_index_patterns.js

```diff
@@ -0,0 +1,21 @@
+import { map, flatten, pluck } from 'lodash';
+
+const getIndexPatternsFromResponse = json => {
+  return map(flatten(pluck(json, 'index_patterns')));
```

Comment:
There should also be some sort of deduplication logic. If you run this script:
```
PUT _template/my_log_template_1
{
    "index_patterns" : "logstash-*",
    "settings" : {
        "number_of_shards" : 1
    }
}

PUT _template/my_log_template_2
{
    "index_patterns" : "logstash-*",
    "settings" : {
        "number_of_shards" : 1
    }
}

PUT _template/my_log_template_3
{
    "index_patterns" : "logstash-*",
    "settings" : {
        "number_of_shards" : 1
    }
}
```
and then type 'log' into the search box, you get three instances of '**log**stash-*'.